bee-message/src/error/packable.rs

```diff
@@ -0,0 +1,195 @@
+);
+impl_wrapped_validated!(MessageUnpackError, InputUnpackError, MessageUnpackError::Input);
+impl_wrapped_validated!(MessageUnpackError, OutputUnpackError, MessageUnpackError::Output);
+impl_wrapped_validated!(MessageUnpackError, PayloadUnpackError, MessageUnpackError::Payload);
+impl_wrapped_validated!(
+    MessageUnpackError,
+    TransactionUnpackError,
+    MessageUnpackError::Transaction
+);
+impl_wrapped_validated!(
+    MessageUnpackError,
+    TransactionEssenceUnpackError,
+    MessageUnpackError::TransactionEssence
+);
+impl_wrapped_validated!(
+    MessageUnpackError,
+    SignatureLockedAssetUnpackError,
+    MessageUnpackError::SignatureLockedAsset
+);
+impl_wrapped_validated!(
+    MessageUnpackError,
+    SignatureLockedSingleUnpackError,
+    MessageUnpackError::SignatureLockedSingle
+);
+impl_wrapped_validated!(
+    MessageUnpackError,
+    UnlockBlockUnpackError,
+    MessageUnpackError::UnlockBlock
+);
+impl_wrapped_validated!(
+    MessageUnpackError,
+    UnlockBlocksUnpackError,
+    MessageUnpackError::UnlockBlocks
+);
+impl_wrapped_variant!(MessageUnpackError, DataUnpackError, MessageUnpackError::Data);
+impl_wrapped_variant!(MessageUnpackError, DkgUnpackError, MessageUnpackError::Dkg);
+impl_wrapped_variant!(MessageUnpackError, FpcUnpackError, MessageUnpackError::Fpc);
+impl_wrapped_variant!(
+    MessageUnpackError,
+    SaltDeclarationUnpackError,
+    MessageUnpackError::SaltDeclaration
+);
+impl_wrapped_variant!(MessageUnpackError, SignatureUnpackError, MessageUnpackError::Signature);
+impl_wrapped_variant!(MessageUnpackError, ValidationError, MessageUnpackError::ValidationError);
+impl_from_infallible!(MessageUnpackError);
+
+impl From<UnpackOptionError<MessageUnpackError>> for MessageUnpackError {
+    fn from(error: UnpackOptionError<MessageUnpackError>) -> Self {
+        match error {
+            UnpackOptionError::Inner(error) => error,
+            UnpackOptionError::UnknownTag(tag) => Self::InvalidOptionTag(tag),
+        }
+    }
+}
+
+impl fmt::Display for MessageUnpackError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Data(e) => write!(f, "error unpacking Data payload: {}", e),
+            Self::Dkg(e) => write!(f, "error unpacking DKG payload: {}", e),
+            Self::Fpc(e) => write!(f, "error unpacking FPC payload: {}", e),
+            Self::Indexation(e) => write!(f, "error unpacking Indexation payload: {}", e),
+            Self::Input(e) => write!(f, "error unpacking Input: {}", e),
+            Self::InvalidPayloadKind(kind) => write!(f, "invalid payload kind: {}.", kind),
+            Self::InvalidOptionTag(tag) => write!(f, "invalid tag for Option: {} is not 0 or 1", tag),
+            Self::Output(e) => write!(f, "error unpacking Output: {}", e),
+            Self::OutputId(e) => write!(f, "error unpacking OutputId: {}", e),
+            Self::Payload(e) => write!(f, "error unpacking payload: {}", e),
```

Comment:
I guess HERE the "p" in payload should be uppercase. Do I now get a prize for noticing this? xD